modules/swagger-codegen/src/main/java/io/swagger/codegen/CodegenOperation.java

```diff
@@ -30,6 +30,23 @@
     public List<Map<String, String>> examples;
     public ExternalDocs externalDocs;
 
+    private boolean nonempty(List<CodegenParameter> params)
+    {
+        return params != null && params.size() > 0;
+    }
+    public boolean getHasBodyParam() {
+        return nonempty(bodyParams);
+    }
+    public boolean getHasQueryParams() {
+        return nonempty(bodyParams);
+    }
+    public boolean getHasHeaderParams() {
+        return nonempty(bodyParams);
+    }
+    public boolean getHasPathParams() {
+        return nonempty(bodyParams);
+    }
+
```

Comment:
Thanks, I was not aware one could use .length in this way (I don't see any uses of that in the existing .mustache files). 

I had seen several `has*` boolean fields (`hasConsumes, hasProduces, hasMore`...) in CodegenOperation.java and wanted to continue that since `{{#hasConsumes}}` seems like a useful pattern. But those fields require initializing in distant classes, which felt wrong. The methods felt like a better implementation.

The property methods I provided are safer in that they work even if the collection is null (most of the other fields are public but not marked final, so there are no guarantees that they are always non-null -- hopefully not, but the method is more robust).